File: 0941-sort-array-by-parity/0941-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int i=0;
        int n = nums.size();
        while(i<n && nums[i]%2==0) i++;
        int j = i+1;
        while(j<n) {
            if(nums[j]%2==0) {
                swap(nums[j],nums[i]);
                i++;
            } 
            j++;
        }
        return nums;
    }
    
};

File: 0941-sort-array-by-parity/0941-sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        left,right=0,0
        n = len(nums)
        while right<n:
            if nums[right]%2==0:
                nums[left],nums[right] = nums[right],nums[left]
                left+=1
            right+=1

        return nums
        